src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -122,6 +123,23 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 
 		this._register(this._onAddConnectionProfile);
 		this._register(this._onDeleteConnectionProfile);
+
+		// keeps azure tokens refreshed every 10 minutes
+		const seconds = 600;
+
+		setInterval(async () => {
+			try {
+				let connectionList = this.getActiveConnections();
+				connectionList.forEach(async connection => {
+					if (connection.authenticationType === AuthenticationType.AzureMFA) {
+						await this.refreshAzureAccountTokens(connection);
```

Comment:
add a try catch block here so that the token refresh for other connections won't be impacted.

Reply:
That technically  isn't necessary - forEach is going to kick off the promises all in a row and they'll each run regardless of what the others are doing. You'll just end up getting some "Unhandled promise rejection" errors if any of them fail. 

But it'd be better to add the try/catch anyways so you can log better error messages (and not have unhandled promise rejection errors). And then you might as well remove the outer try/catch since the only thing that could fail on is getActiveConnections which should never be expected to fail (and if it does it doesn't matter)